src/components/molecules/Visualizer/Engine/Cesium/Box/index.tsx

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import {
+  Axis,
+  CallbackProperty,
+  Cartesian2,
+  Cartesian3,
+  Color,
+  Matrix4,
+  Plane as CesiumPlane,
+  PositionProperty,
+  TranslationRotationScale,
+} from "cesium";
+import React, { useMemo, FC, useEffect, memo, useState } from "react";
+import { Entity, PlaneGraphics } from "resium";
+
+import { LatLng, toColor } from "@reearth/util/value";
+
+import type { Props as PrimitiveProps } from "../../../Primitive";
+
+export type Props = PrimitiveProps<Property>;
+
+export type BoxStyle<C = Color | string> = {
+  fillColor?: C;
+  outlineColor?: C;
+  outlineWidth?: number;
+  fill?: boolean;
+  outline?: boolean;
+};
+
+export type Property = {
+  default?: {
+    position?: LatLng;
+    location?: LatLng;
+    height?: number;
+    dimensions?: {
+      x: number;
+      y: number;
+      z: number;
+    };
+  } & BoxStyle<string>;
+};
+
+// ref: https://github.com/TerriaJS/terriajs/blob/cad62a45cbee98c7561625458bec3a48510f6cbc/lib/Models/BoxDrawing.ts#L1446-L1461
+function setPlaneDimensions(
+  boxDimensions: Cartesian3,
+  planeNormalAxis: Axis,
+  planeDimensions: Cartesian2,
+) {
+  if (planeNormalAxis === Axis.X) {
+    planeDimensions.x = boxDimensions.y;
+    planeDimensions.y = boxDimensions.z;
+  } else if (planeNormalAxis === Axis.Y) {
+    planeDimensions.x = boxDimensions.x;
+    planeDimensions.y = boxDimensions.z;
+  } else if (planeNormalAxis === Axis.Z) {
+    planeDimensions.x = boxDimensions.x;
+    planeDimensions.y = boxDimensions.y;
+  }
+}
+
+const Side: FC<{
```

Comment:
Can you split Side in to another file and then import it in here? There is a lot of logic between Side and Box and it would be nice to keep things a little more segregated.

Reply:
I fixed in [refactor: split Side to other file](https://github.com/reearth/reearth-web/pull/338/commits/8305e3060d4ceee9b701b2029b32085b127baef9)